src/common/cid.ts

```diff
@@ -0,0 +1,26 @@
+import { CID } from "multiformats/cid"
+
+
+/**
+ * Decode a possibly string-encoded CID.
+ * Passing an already decoded CID instance works too.
+ * NOTE: Throws an error if a CID cannot be decoded!
+ */
+export function decodeCID(val: CID | object | string): CID {
+  const cid = CID.asCID(val)
+  if (cid) return cid
+
+  if (typeof val === "string") return CID.parse(val as string)
+  if (typeof val === "object" && "version" in val && "code" in val && "multihash" in val) {
+    return CID.create(val.version, val.code, val.multihash)
+  }
```

Comment:
What do you think about adding these helper functions I've been using a lot in various projects so far?
```ts
export function hasProp<K extends PropertyKey>(data: unknown, prop: K): data is Record<K, unknown> {
  return typeof data === "object" && data != null && prop in data
}
export function isRecord(data: unknown): data is Record<PropertyKey, unknown> {
  return typeof data === "object" && data != null
}
```
Then you'd use `isRecord(val) && hasProp(val, "version") && hasProp(val, "code") && hasProp(val, "multihash")`.
I think typescript doesn't have support for `"string" in x` to actually do anything to the types as far as I know (correct me if I'm wrong).

Also there's a couple of gotchas with `typeof val === "object"`, because e.g. `typeof null === "object"`, so I like getting used to using `isRecord`.

Reply:
We already have the `hasProp` and `isObject` functions yeah. That said, Typescript does support the `in` operator: https://www.typescriptlang.org/docs/handbook/2/narrowing.html#the-in-operator-narrowing